src/downloadAzureProject/setupProjectFolder.ts

```diff
@@ -0,0 +1,115 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { WebSiteManagementModels } from '@azure/arm-appservice';
+import { RequestPrepareOptions } from '@azure/ms-rest-js';
+import * as extract from 'extract-zip';
+import * as querystring from 'querystring';
+import * as vscode from 'vscode';
+import { callWithTelemetryAndErrorHandling, IActionContext } from 'vscode-azureextensionui';
+import { ProjectLanguage } from '../constants';
+import { GlobalStates } from '../extension';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { SlotTreeItemBase } from "../tree/SlotTreeItemBase";
+import { getNameFromId } from '../utils/azure';
+import { requestUtils } from '../utils/requestUtils';
+
+export async function setupProjectFolder(uri: vscode.Uri, vsCodeFilePathUri: vscode.Uri): Promise<void> {
+    const parsedQuery: querystring.ParsedUrlQuery = querystring.parse(uri.query);
+    const resourceId: string = getRequiredQueryParameter(parsedQuery, 'resourceId');
+    const devContainerName: string = getRequiredQueryParameter(parsedQuery, 'devcontainer');
+    const language: string = getRequiredQueryParameter(parsedQuery, 'language');
+    const downloadAppContent: string = getRequiredQueryParameter(parsedQuery, 'downloadAppContent');
+
+    ext.context.globalState.update(GlobalStates.projectLanguage, getProjectLanguageForLanguage(language));
+    const toBeDeletedFolderPathUri: vscode.Uri = vscode.Uri.joinPath(vsCodeFilePathUri, 'temp');
+
+    try {
+        const functionAppName: string = getNameFromId(resourceId);
+        const downloadFilePath: string = vscode.Uri.joinPath(toBeDeletedFolderPathUri, `${functionAppName}.zip`).fsPath;
+
+        vscode.window.showInformationMessage(localize('settingUpFunctionAppLocalProjInfoMessage', `Setting up project for function app '${functionAppName}' with language '${language}'.`));
```

Comment:
This would work better than `showInformationMessage`:

`vscode.window.withProgress({ location: vscode.ProgressLocation.Notification`